src/Framework/PaymentGateways/Actions/GenerateGatewayRouteUrl.php

```diff
@@ -2,18 +2,24 @@
 
 namespace Give\Framework\PaymentGateways\Actions;
 
+use Give\Framework\PaymentGateways\DataTransferObjects\GatewayRouteData;
+use InvalidArgumentException;
+
 class GenerateGatewayRouteUrl
 {
     /**
      * @since 2.18.0
+     * @unreleased Return URL with nonce action.
      *
-     * @param  string  $gatewayId
-     * @param  string  $gatewayMethod
-     * @param  int  $donationId
-     * @param  array|null  $args
-     * @return string
+     * @param string $gatewayId
+     * @param string $gatewayMethod
+     * @param int $donationId
+     * @param array|null $additionalQueryArgs
+     * @param array|null $optParams
+     *
+     * @return string Url.
      */
-    public function __invoke($gatewayId, $gatewayMethod, $donationId, $args = null)
+    public function __invoke($gatewayId, $gatewayMethod, $donationId, $additionalQueryArgs = null, $optParams = null)
```

Comment:
As previously commented, we need to think through properly including & validating nonces.  This would not be clear enough if it's something we need to do.